### PR TITLE
Add logic and content to GCSE components for UCL courses

### PR DESCRIPTION
**Pull request by george-bit**
### Context
In order to render different reqs for ucl courses I've added a method on
the course model to determine which courses are ucl. Added logic to the
view components to render different reqs in this case.

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased master
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by duncanjbrown** [comment]
Discussed with @george-bit and @paulrobertlloyd — we're going to move this information into TTAPI and stay dumb out here in Publish

**Comment by duncanjbrown** [comment]
TTAPI PR: https://github.com/DFE-Digital/teacher-training-api/pull/2018

File: app/components/gcse_requirements_component.rb [hunk 1]
```diff
@@ -8,11 +8,18 @@ def initialize(course:)
...
+      5 => "5 (C)",
+      4 => "4 (C)",
+    }
+
+    grade = grade_mapping.fetch(course.gcse_grade_required)
+
     case course.level
     when "primary"
-      "Grade 4 (C) or above in English, maths and science, or equivalent qualification"
+      "Grade #{grade} or above in English, maths and science, or equivalent qualification"
```
Comment (defong) [inline app/components/gcse_requirements_component.rb:20]:
```suggestion
      "Grade #{course.gcse_grade_required} (C) or above in English, maths and science, or equivalent qualification"
```

File: app/components/gcse_requirements_component.rb [hunk 2]
```diff
@@ -8,11 +8,18 @@ def initialize(course:)
...
     when "secondary"
-      "Grade 4 (C) or above in English and maths, or equivalent qualification"
+      "Grade #{grade} or above in English and maths, or equivalent qualification"
```
Comment (defong) [inline app/components/gcse_requirements_component.rb:22]:
```suggestion
      "Grade #{course.gcse_grade_required} (C) or above in English and maths, or equivalent qualification"
```

**Review by defong — APPROVED** [review]

**Comment by defong** [comment]
For clarify

> TTAPI PR: [DFE-Digital/teacher-training-api#2018](https://github.com/DFE-Digital/teacher-training-api/pull/2018)


"UCL, University College London (University of London) (U80) [2021/22]"
"University College London (UCL) (I30) [2021/22]"

@duncanjbrown half of it is pointless

image

See https://trello.com/c/LANwfOEG/1665-fix-duplicate-providers from 

u80

**Pull request merged**